### ISSUE TITLE
"completed" tag not properly applying to files since 10.0.1

### ISSUE THREAD

**Issue by vormav78** [UniversalMediaServer/UniversalMediaServer#2503]
Hello, I don't know if this is the right place to report the problem but I couldn't find a similar post so here I am.

The tag that says a file has been completed (replay circle) is no longer working since 10.0.1 (windows)
I have also tried 10.5.0 and I have the same issue.

As soon as I install 9.8.0 (clean install) it fixes the problem. Note that the marked files under 9.8.0 stay marked when I upgrade. It just seems the tag is no longer properly applied.

**Comment by SubJunk** [comment]
@vormav78 can you please provide logs for that? I have seen this bug reported by other people too, but it is working for me when I test it, so I need logs to help

**Comment by SubJunk** [comment]
Instructions for logs are at https://www.universalmediaserver.com/forum/viewtopic.php?f=9&t=556

**Comment by vormav78** [comment]
Sure, I can reinstall the newest version and reproduce the issue. Do you need the trace to be done on multiple files or will one file be enough?

**Comment by vormav78** [comment]
Here are the logs. I found out that, on a clean install before I personalize my settings, it actually works as intended. I'm suspecting one of the other options has this unintended side effect but I am still trying to find which one that is as I leave "Fully played action" to "Mark Media".

[ums_dbg_2021-05-27-00-12.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/6551097/ums_dbg_2021-05-27-00-12.zip)

**Comment by SubJunk** [comment]
Thanks for the logs and notes, that's very helpful. Also you mentioned 10.0.1, does that mean it works properly in 10.0.0?

**Comment by SubJunk** [comment]
Can you try enabling the Media Library? It might have got mixed up with that logic somehow in v10

**Comment by vormav78** [comment]
I have not tried 10.0.0  but I reverted back to the last version I had installed that worked which was 9.8.0. 

Alright so after at least 5 hours of testing different permutations of settings I think I found the culprit: It seems the function stops working whenever I disable "Enable Media Library". It took me a long time to find since that setting has no reason to even touch such a configuration but I did it! I hope this helps :)

**Comment by SubJunk** [comment]
@vormav78 nice one, thanks for testing. I also realised that that is expected behaviour and it is my fault for using bad naming for the settings :( I'm sorry for wasting your time on this. I'll try to explain now:

In the past (in versions up to v6), the "Enable the Media Library" setting was called "Enable cache", but I renamed it when I started expanding the caching to include more features like the IMDb filtering it now has.
In v10 I made the fully played information get stored in the cache (AKA the local database) because that lets the program include that information in its dynamic folders, like the ability to view all unwatched movies.

But I didn't rename that "Enable the Media Library" setting at that point, and I should have. So when you uncheck that setting you are really telling UMS "do not store any information in your cache/database, including fully played status of files".

I will fix this in the next release by renaming the setting and giving it a tooltip to explain these things.

Again I'm sorry for the confusion.
